pre-docs/writing_plans.md

```diff
@@ -474,3 +474,30 @@ plan run_with_facts(TargetSpec $nodes) {
   run_task(ubuntu_task, $ubuntu_nodes)
 }
 ```
+
+If the targets are managed by puppet and bolt is (configured to connect to
+puppetdb)[bolt_configuring_puppetdb.md] you can use the puppetdb_facts plan
+to query facts from puppetdb and store them in the inventory.
+
+```puppet
+run_plan('puppetdb_fact', nodes => $nodes)
+```
+
+## `puppetdb_query`
```

Comment:
It might be useful to have a "Supporting functions" or "Data functions" heading for things like `vars`, `facts`, and `puppetdb_query`, probably after the "Plan execution functions" section.